File: 59.promedio_pares/promedio_pares.py
# Ejercicio No. 59: Promedio de pares.

# processing
def promediar_pares(lista):
    suma = 0
    promedio = 0
    cont = 0
    for i in lista:
        if i&2 == 0:
            suma += i
            cont += 1
    promedio = suma / cont
    return promedio

lista1 = [1,2,3,4,5]
promedio = promediar_pares(lista1)

# output
print(promedio)

File: 59.promedio_pares/PromedioPares.java
package PromedioPares;

import javax.swing.JOptionPane;

public class PromedioPares {
    //Processing
   public static double promediarPares(int[] lista){
    double suma = 0;
    double promedio = 0.0;
    int cont = 0;

    for(int i = 0; i<lista.length; i++){
        if(lista[i]%2 == 0){
            suma += lista[i];
            cont += 1;
        }
    }

    promedio = suma / cont;
    return promedio;

   }
   public static void main(String[] args) {

    int[] lista1 = {1, 2, 3, 4, 5};
    double promedio = promediarPares(lista1);

    //Output
    JOptionPane.showMessageDialog(null, promedio);
    System.exit(0);
    
   }
}